### ISSUE TITLE
Reset doesn't clear DataStoreSupport encryption key

### ISSUE THREAD

**Issue by sashei** [mozilla-lockwise/lockwise-android#282]
When receiving `DataStoreAction.Reset`, we don't clear & regenerate the key that the `LoginsStorage` uses to encrypt the database on disk.

**Comment by sandysage** [comment]
@sashei Does this need to happen prior to internal Alpha launch? Is there a pressing reason for this to be in the backlog or can we mark as a P2 to address after the internal release?

**Comment by linuxwolf** [comment]
FWIW, I don't think this has to happen before alpha.  It goes to truly cleaning up, but I don't see a critical security concern here.

**Comment by sashei** [comment]
👍 agree with matt

**Comment by eliserichards** [comment]
Related to #649. We can repurpose it as a followup for that issue.

**Comment by sashei** [comment]
I think we can close as a duplicate of #649. the work is the same.
